source/MessageHandlers/SendDataHandler.ts

```diff
@@ -74,29 +80,36 @@ export default class SendDataHandler implements IMessageHandler {
 
         return new Promise<void>((resolve, reject) => {
             let completionDescription: string;
+            const validTables = {};
+            for (const tbl of integrationConfig.queries.map((q) => q.name)) {
+                validTables[tbl] = tbl;
+            }
+
             mapLimit(integrationConfig.queries, STATEMENT_CONCURRENCY,
-                async (queryStatement: IQueryDefinition, queryCallback: any) => { // iterator value callback
+                async (queryDefinition: IQueryDefinition, queryCallback: any) => { // iterator value callback
                     let reader: IDataReader;
                     let itemMetadata: Readable;
+
                     try {
                         const startTime = Date.now();
 
                         reader = this._container.get<IDataReader>(TYPES.DataReader);
-                        const readable: IQueryResult = await reader.read(queryStatement.query);
-                        itemMetadata = readable.metadata;
-                        await this._writer.ingest(readable.result, folderPath, queryStatement.name);
+                        const queryResult: IQueryResult = await reader.read(queryDefinition);
+                        itemMetadata = queryResult.metadata;
+                        await this._writer.ingest(queryResult.result, folderPath, queryDefinition.name);
                         const endTime = Date.now();
                         const diff = moment.duration(endTime - startTime);
                         completionDescription = diff.humanize(false);
                         this._logger.info(
-                            `Ingestion for '${queryStatement.name}' ` +
+                            `Ingestion for '${queryDefinition.name}' ` +
                             `took ${completionDescription} (${diff.asMilliseconds()}ms)`
                         );
                     } catch (err) {
+                        delete validTables[queryDefinition.name];
```

Comment:
Various failures may directly or indirectly indicate a missing table so we must remove it from the list of known valid tables. In the end we're left with valid tables we can reliable go fetch DDL for.